From b964b4295bcef2352e554e2d22120e759640ee60 Mon Sep 17 00:00:00 2001
From: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Date: Mon, 3 Mar 2025 09:53:16 +0100
Subject: [PATCH 1623/1695] drivers: stm32-core: use DIV_ROUND_UP instead
 ROUNDUP_DIV

Implementation of stm32_div_get_rate() in MP13, MP2x clock driver
uses ROUNDUP_DIV() macro with divisor arguments that may not be
a power of 2 despite the macro expects it is.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git ac2bd8774ba4a4c74adc18ed7c16c02c3c5803d4 drivers: stm32-core: use DIV_ROUND_UP instead ROUNDUP_DIV]
Signed-off-by: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Change-Id: I0f058361907ba4f858ec7e9c2900849582b75b51
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/441046
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/drivers/clk/clk-stm32-core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/drivers/clk/clk-stm32-core.c b/core/drivers/clk/clk-stm32-core.c
index d1c634976..7783d960d 100644
--- a/core/drivers/clk/clk-stm32-core.c
+++ b/core/drivers/clk/clk-stm32-core.c
@@ -329,7 +329,7 @@ unsigned long stm32_div_get_rate(int div_id, unsigned long prate)
 	if (!div)
 		return prate;
 
-	return ROUNDUP_DIV((uint64_t)prate, div);
+	return DIV_ROUND_UP((uint64_t)prate, div);
 }
 
 TEE_Result stm32_div_set_rate(int div_id, unsigned long rate,
-- 
2.47.2

